### PR TITLE
Centrality: Week 1

### PR DESCRIPTION
**Pull request by bedupako12mas**
Changes proposed in this pull request:

- [x] Added some of the necessary files for implementing ```pgr_centrality```.
  - [x] ```doc/metrics/``` - CMakeLists.txt,  metrics-family.rst,  pgr_centrality.rst
  - [x] ```docqueries/metrics``` - CMakeLists.txt,  doc-centrality.result,  doc-centrality.test.sql,  test.conf
  - [x] ```sql/metrics``` - CMakeLists.txt, _centrality.sql,  centrality.sql  
  - [x] ```src/metrics``` - CMakeLists.txt, centrality.c, centrality_driver.cpp,  
- [x] Modified the files in the sql/ and src/ directories (licenses and function names)
- [x] Added metrics to configuration.conf

### CODE REVIEW

File: doc/metrics/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+
+SET(LOCAL_FILES
+    allpairs-family.rst
```
Comment (cvvergara) [inline doc/metrics/CMakeLists.txt:3]: TODO
```
metrics-family.rst
pgr_centrality.rst
```

File: doc/metrics/metrics-family.rst [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+    This documentation is licensed under a Creative Commons Attribution-Share
+    Alike 3.0 License: https://creativecommons.org/licenses/by-sa/3.0/
+   ****************************************************************************
+
+|
+
+
+
+All Pairs - Family of Functions
```
Comment (cvvergara) [inline doc/metrics/metrics-family.rst:14]: Fix

File: doc/metrics/metrics-family.rst [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+===============================================================================
+
+The following functions work on all vertices pair combinations
+
+.. index from here
+
+* :doc:`pgr_floydWarshall` - Floyd-Warshall's algorithm.
```
Comment (bedupako12mas) [inline doc/metrics/metrics-family.rst:21]: Fix

File: doc/metrics/metrics-family.rst [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+* :doc:`pgr_johnson` - Johnson's algorithm
+
+.. index to here
+
+.. toctree::
+    :hidden:
+
+    pgr_floydWarshall
```
Comment (bedupako12mas) [inline doc/metrics/metrics-family.rst:29]: Fix

File: doc/metrics/metrics-family.rst [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    pgr_johnson
+
+Introduction
+-------------------------------------------------------------------------------
+
+.. characteristics_start
```
Comment (bedupako12mas) [inline doc/metrics/metrics-family.rst:35]: Fix characteristics

File: doc/metrics/metrics-family.rst [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+-------------------------------------------------------------------------------
+
+Edges SQL
+...............................................................................
+
+.. include:: pgRouting-concepts.rst
+    :start-after: no_id_edges_sql_start
+    :end-before: no_id_edges_sql_end
+
+Result columns
```
Comment (bedupako12mas) [inline doc/metrics/metrics-family.rst:100]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+    Copyright(c) pgRouting Contributors
+
+    This documentation is licensed under a Creative Commons Attribution-Share
+    Alike 3.0 License: https://creativecommons.org/licenses/by-sa/3.0/
+   ****************************************************************************
+
+|
+
+
+``pgr_floydWarshall``
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:13]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+===============================================================================
+
+``pgr_floydWarshall`` - Returns the sum of the costs of the shortest path for
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:16]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+each pair of nodes in the graph using Floyd-Warshall algorithm.
+
+.. figure:: images/boost-inside.jpeg
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:19]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 10]
```diff
@@ -0,0 +1,109 @@
...
+   :target: https://www.boost.org/libs/graph/doc/floyd_warshall_shortest.html
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:20]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+* Version 2.2.0
+
+  * Signature change
+  * Old signature no longer supported
+
+* Version 2.0.0
+
+  * **Official** function
+
+Description
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:35]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+
+``pgr_floydWarshall`` - Returns the sum of the costs of the shortest path for
+each pair of nodes in the graph using Floyd-Warshall algorithm.
+
+.. figure:: images/boost-inside.jpeg
+   :target: https://www.boost.org/libs/graph/doc/floyd_warshall_shortest.html
+
+   Boost Graph Inside
+
+.. rubric:: Availability
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:24]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  * Old signature no longer supported
+
+* Version 2.0.0
+
+  * **Official** function
+
+Description
+-------------------------------------------------------------------------------
+
+The Floyd-Warshall algorithm, also known as Floyd's algorithm, is a good choice
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:38]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+to calculate the sum of the costs of the shortest path for each pair of nodes in
+the graph, for *dense graphs*.
+We use Boost's implementation which runs in :math:`\Theta(V^3)` time,
+
+.. include:: allpairs-family.rst
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:43]: Remove

File: doc/metrics/pgr_centrality.rst [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+   :start-after: characteristics_start
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:44]: remove

File: doc/metrics/pgr_centrality.rst [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+
+Signatures
+-------------------------------------------------------------------------------
+
+.. rubric:: Summary
+
+.. admonition:: \ \
+   :class: signatures
+
+   pgr_floydWarshall(`Edges SQL`_, [``directed``])
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:55]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+   | Returns set of |matrix-result|
+   | OR EMPTY SET
+
+
+:Example: For a directed subgraph with edges :math:`\{1, 2, 3, 4\}`.
+
+.. literalinclude:: doc-floydWarshall.queries
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:63]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+   :start-after: -- q1
+   :end-before: -- q2
+
+Parameters
+-------------------------------------------------------------------------------
+
+.. include:: allpairs-family.rst
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:70]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+    :start-after: dijkstra_optionals_start
+    :end-before: dijkstra_optionals_end
+
+Inner Queries
+-------------------------------------------------------------------------------
+
+Edges SQL
+...............................................................................
+
+.. include:: pgRouting-concepts.rst
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:87]: fix

File: doc/metrics/pgr_centrality.rst [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    :start-after: no_id_edges_sql_start
+    :end-before: no_id_edges_sql_end
+
+Result columns
+-------------------------------------------------------------------------------
+
+.. include:: pgRouting-concepts.rst
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:94]: Fix

File: doc/metrics/pgr_centrality.rst [hunk 21]
```diff
@@ -0,0 +1,109 @@
...
+    :start-after: return_cost_start
+    :end-before: return_cost_end
+
+See Also
```
Comment (bedupako12mas) [inline doc/metrics/pgr_centrality.rst:98]: Remove

**Pull request merged**